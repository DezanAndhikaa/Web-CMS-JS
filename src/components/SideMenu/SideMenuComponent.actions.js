

export const CLICK_MENU = 'CLICK_MENU';
export const LOGOUT_ACTION = 'USER_LOGOUT';

export function clickMenuAction(payload) {
	return { type: CLICK_MENU, payload };
}

export function logoutAction() {
	return { type: LOGOUT_ACTION };
}
