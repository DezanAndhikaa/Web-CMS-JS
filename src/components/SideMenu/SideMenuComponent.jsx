import React from "react";
import {
  ListItemText,
  ListItemIcon,
  ListItem,
  List,
  Drawer,
  Modal,
  DialogContent,
  Collapse
} from "@material-ui/core";
import { UTLogoNew} from "../../assets/imgs/";
import { AssignmentIcon, LogoutIcon, AllocationIcon, DeliveryIcon, TrackingIcon,
SettingIcon, ProductionIcon, ExecutionIcon, DashboardIcon, PlanningIcon } from "../../assets/icons";
import { Menu, StorageKey } from "../../constants";
import LogoutModal from "./Logout";
import "./SideMenuComponent.scss";

class SideMenuComponent extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isLogoutModalShown: false
    };
  }

  handleLogout = () => {
    localStorage.removeItem(StorageKey.USER_DATA);
    this.props.logout();
    this.props.onLogout();
    this.setState({ isLogoutModalShown: false });
    this.props.push(Menu.LOGIN);
  };

  handleClick(menu, subMenu) {
    console.log('skuit router menu', menu)
    console.log('skuit router sub', subMenu)
    this.props.clickMenu(menu, subMenu);
    if (menu === Menu.PLANNING) {
      if (subMenu !== "") this.props.push(subMenu);
    }else if (menu === Menu.BACKLOG){
      if (subMenu !== "") this.props.push(subMenu);
    } else {
      this.props.push(menu);
    }
  }

  render() {
    let drawer = null;
    if (
      this.props.path !== Menu.LOGIN &&
      this.props.path !== `${Menu.LOGIN}/`
    ) {
      drawer = (
        <Drawer
          className="menu"
          variant={this.props.displayMode === "web" ? "permanent" : "temporary"}
          classes={{ paper: "drawer-paper", docked: "docked" }}
          anchor="left"
          open={this.props.menuDrawerState}
          onClose={() => this.props.closeDrawer()}
        >
          <List>
            {/* Dashboard */}
            <ListItem
              button
              key="dashboard"
              className={
                this.props.path === Menu.DASHBOARD
                  ? "menu-item-selected"
                  : "menu-item"
              }
              onClick={() => this.handleClick(Menu.DASHBOARD, "")}
            >
              <ListItemIcon classes={{ root: "icon-root" }}>
                <img
                  src={DashboardIcon}
                  alt="assignment icon"
                  className="item-icon"
                />
              </ListItemIcon>
              <ListItemText
                primary="Dashboard"
                classes={{ primary: "item-text", root: "item-text" }}
              />
            </ListItem>
            {/* Tracking */}
            <ListItem
              button
              key="jobs"
              className={
                this.props.path.includes(Menu.PROBLEMLOG)
                  ? "menu-item-selected"
                  : "menu-item"
              }
              onClick={() => this.handleClick(Menu.PROBLEMLOG, "")}
            >
              <ListItemIcon classes={{ root: "icon-root" }}>
                <img
                  src={TrackingIcon}
                  alt="assignment icon"
                  className="item-icon"
                />
              </ListItemIcon>
              <ListItemText
                primary="Tracking"
                classes={{ primary: "item-text", root: "item-text" }}
              />
            </ListItem>
            {/* Planning */}
            <ListItem
              button
              key="plans"
              className={
                this.props.path.includes(Menu.PLANNING)
                  ? "menu-item-selected"
                  : "menu-item"
              }
              onClick={() => this.handleClick(Menu.PLANNING, "")}
            >
              <ListItemIcon classes={{ root: "icon-root" }}>
                <img
                  src={PlanningIcon}
                  alt="assignment icon"
                  className="item-icon"
                />
              </ListItemIcon>
              <ListItemText
                primary="Planning"
                classes={{ primary: "item-text", root: "item-text" }}
              />
            </ListItem>
            <Collapse
              in={this.props.plansMenuExpanded}
              timeout="auto"
              unmountOnExit
            >
              <List disablePadding>
                <ListItem
                  button
                  key="plans-assignment"
                  className={
                    this.props.path === Menu.PLANNING_DASHBOARD
                      ? "sub-menu-selected"
                      : "sub-menu"
                  }
                  onClick={() => this.handleClick(Menu.PLANNING, Menu.PLANNING_DASHBOARD)}
                >
                  <ListItemIcon classes={{ root: "icon-root" }}>
                    <img
                      src={AssignmentIcon}
                      alt="assignment icon"
                      className="item-icon"
                    />
                  </ListItemIcon>
                  <ListItemText
                    primary="Dashboard"
                    classes={{ primary: "item-text", root: "item-text" }}
                  />
                </ListItem>
                <ListItem
                  button
                  key="jobs-report"
                  className={
                    this.props.path === Menu.PLANNING_DETAILS
                      ? "sub-menu-selected"
                      : "sub-menu"
                  }
                  onClick={() => this.handleClick(Menu.PLANNING, Menu.PLANNING_DETAILS)}
                >
                  <ListItemIcon classes={{ root: "icon-root" }}>
                    <img
                      src={AssignmentIcon}
                      alt="assignment icon"
                      className="item-icon"
                    />
                  </ListItemIcon>
                  <ListItemText
                    primary="Detail"
                    classes={{ primary: "item-text", root: "item-text" }}
                  />
                </ListItem>
              </List>
            </Collapse>
            {/* Production */}
            <ListItem
              button
              key="jobs"
              className={
                this.props.path.includes(Menu.DETAIL_PI)
                  ? "menu-item-selected"
                  : "menu-item"
              }
              onClick={() => this.handleClick(Menu.DETAIL_PI, "")}
            >
              <ListItemIcon classes={{ root: "icon-root" }}>
                <img
                  src={ProductionIcon}
                  alt="assignment icon"
                  className="item-icon"
                />
              </ListItemIcon>
              <ListItemText
                primary="Production"
                classes={{ primary: "item-text", root: "item-text" }}
              />
            </ListItem>
            {/* Allocation */}
            <ListItem
              button
              key="jobs"
              className={
                this.props.path.includes(Menu.ALLOCATION)
                  ? "menu-item-selected"
                  : "menu-item"
              }
              onClick={() => this.handleClick(Menu.ALLOCATION, "")}
            >
              <ListItemIcon classes={{ root: "icon-root" }}>
                <img
                  src={AllocationIcon}
                  alt="assignment icon"
                  className="item-icon"
                />
              </ListItemIcon>
              <ListItemText
                primary="Allocation"
                classes={{ primary: "item-text", root: "item-text" }}
              />
            </ListItem>
            {/* Delivery */}
            <ListItem
              button
              key="jobs"
              className={
                this.props.path.includes(Menu.FC)
                  ? "menu-item-selected"
                  : "menu-item"
              }
              onClick={() => this.handleClick(Menu.FC, "")}
            >
              <ListItemIcon classes={{ root: "icon-root" }}>
                <img
                  src={DeliveryIcon}
                  alt="assignment icon"
                  className="item-icon"
                />
              </ListItemIcon>
              <ListItemText
                primary="Delivery"
                classes={{ primary: "item-text", root: "item-text" }}
              />
            </ListItem>
            {/* Execution */}
            <ListItem
              button
              key="jobs"
              className={
                this.props.path.includes(Menu.MASTER_DATA)
                  ? "menu-item-selected"
                  : "menu-item"
              }
              onClick={() => this.handleClick(Menu.MASTER_DATA, "")}
            >
              <ListItemIcon classes={{ root: "icon-root" }}>
                <img
                  src={ExecutionIcon}
                  alt="assignment icon"
                  className="item-icon"
                />
              </ListItemIcon>
              <ListItemText
                primary="Execution"
                classes={{ primary: "item-text", root: "item-text" }}
              />
            </ListItem>
            {/* Setting */}
            <ListItem
              button
              key="settings"
              className={
                this.props.path === Menu.SETTINGS
                  ? "menu-item-selected"
                  : "menu-item"
              }
              onClick={() => this.handleClick(Menu.SETTINGS, "")}
            >
              <ListItemIcon classes={{ root: "icon-root" }}>
                <img
                  src={SettingIcon}
                  alt="assignment icon"
                  className="item-icon"
                />
              </ListItemIcon>
              <ListItemText
                primary="Settings"
                classes={{ primary: "item-text", root: "item-text" }}
              />
            </ListItem>
            <ListItem />
            <ListItem
              button
              key="logout"
              className={
                this.props.path === Menu.LOGOUT
                  ? "menu-item-selected"
                  : "menu-item"
              }
              onClick={() => this.setState({ isLogoutModalShown: true })}
            >
              <ListItemIcon classes={{ root: "icon-root" }}>
                <img
                  src={LogoutIcon}
                  alt="assignment icon"
                  className="item-icon"
                />
              </ListItemIcon>
              <ListItemText
                primary="Logout"
                classes={{ primary: "item-text", root: "item-text" }}
              />
            </ListItem>
          </List>
          <Modal
            open={this.state.isLogoutModalShown}
            className="modal-container"
            onClose={() => this.setState({ isLogoutModalShown: false })}
          >
            <DialogContent className="modal-content">
              <LogoutModal
                onYesClicked={this.handleLogout}
                onNoClicked={() => this.setState({ isLogoutModalShown: false })}
              />
            </DialogContent>
          </Modal>
        </Drawer>
      );
    }
    return drawer;
  }
}

export default SideMenuComponent;
