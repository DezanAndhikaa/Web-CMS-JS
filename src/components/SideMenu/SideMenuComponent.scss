@import '../../themes/variables';
@import '../../themes/functions';

@media (min-width: 1025px){
    .menu{
        flex-shrink: 0;
        .drawer-paper{
            border: none;
            box-shadow: 0 vw(15) vw(30) rgba(0, 0, 0, 0.161);
            width: vw(281) !important;
            margin-top: vw(88);
            .menu-item{
                height: vw(37);
                margin-bottom: vw(10);
                margin-left: vw(12);
                border-radius: vw(15);
                padding-left: vw(32);
                .item-text{
                    font-family: 'Lato-Bold' !important;
                    font-style: normal;
                    font-size: vw(15);
                    line-height: vw(18);
                    padding: 0 !important;
                }
                .item-icon{
                    height: vw(16.31);
                }
            }
            .menu-item:hover{
                background-color: map-get($colors , primary);
            }
            .icon-root{
                min-width: vw(10);
                margin-right: vw(10);
            }
            .menu-item-selected{
                height: vw(37);
                margin-bottom: vw(10);
                margin-left: vw(50);
                background-color: map-get($colors , primary);
                border-radius: vw(15);
                padding-left: vw(32);
                .item-text{
                    font-family: 'Lato-Bold' !important;
                    font-style: normal;
                    font-size: vw(15);
                    line-height: vw(18);
                    padding: 0 !important;
                }
                .item-icon{
                    height: vw(16.31);
                }
            };
            .sub-menu{
                margin-left: vw(75);
                height: vw(37);
                margin-bottom: vw(10);
                .item-text{
                    font-family: 'Lato-Bold' !important;
                    font-style: normal;
                    font-size: vw(15);
                    line-height: vw(18);
                    padding: 0 !important;
                }
                .item-icon{
                    height: vw(16.31);
                }
            }
            .sub-menu-selected{
                margin-left: vw(75);
                height: vw(37);
                margin-bottom: vw(10);
                .item-text{
                    font-family: 'Lato-Bold' !important;
                    font-size: vw(15);
                    line-height: vw(18);
                    padding: 0 !important;
                    color: map-get($colors , primary);
                }
                .item-icon{
                    height: vw(16.31);
                }
            }
        }
    }
}

@media (max-width: 1025px) {
    .menu{
        flex-shrink: 0;
        .drawer-paper{
            width: vw-tab(310) !important;
            border: none;
            .dca-logo{
                width: vw-tab(184);
                height: vw-tab(45);
                margin: vw-tab(24) auto vw-tab(50) auto;
            }
            .menu-item{
                height: vw-tab(37);
                margin-bottom: vw-tab(10);
                padding-left: vw-tab(63);
                .item-text{
                    font-family: 'Lato-Bold' !important;
                    font-style: normal;
                    font-size: vw-tab(15);
                    line-height: vw-tab(18);
                    padding: 0 !important;
                }
                .item-icon{
                    height: vw-tab(16.31);
                }
            }
            .menu-item:hover{
                background-color: map-get($colors , primary);
            }
            .icon-root{
                min-width: 1vw;
                margin-right: vw-tab(10);
            }
            .menu-item-selected{
                height: vw-tab(37);
                margin-bottom: vw-tab(10);
                background-color: map-get($colors , primary);
                padding-left: vw-tab(63);
                .item-text{
                    font-family: 'Lato-Bold' !important;
                    font-style: normal;
                    font-size: vw-tab(15);
                    line-height: vw-tab(18);
                    padding: 0 !important;
                }
                .item-icon{
                    height: vw-tab(16.31);
                }
            };
            .sub-menu{
                padding-left: vw-tab(99);
                height: vw-tab(37);
                margin-bottom: vw-tab(10);
                .item-text{
                    font-family: 'Lato-Bold' !important;
                    font-style: normal;
                    font-size: vw-tab(15);
                    line-height: vw-tab(18);
                    padding: 0 !important;
                }
                .item-icon{
                    height: vw-tab(16.31);
                }
            }
            .sub-menu-selected{
                padding-left: vw-tab(99);
                height: vw-tab(37);
                margin-bottom: vw-tab(10);
                .item-text{
                    font-family: 'Lato-Bold' !important;
                    font-size: vw-tab(15);
                    line-height: vw-tab(18);
                    padding: 0 !important;
                    color: map-get($colors , primary);
                }
                .item-icon{
                    height: vw-tab(16.31);
                }
            }
        }
    }
}