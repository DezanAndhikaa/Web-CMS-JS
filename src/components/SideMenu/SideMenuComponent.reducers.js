import { Menu } from 'constants/index';
import { CLICK_MENU } from './SideMenuComponent.actions';

const initialState = {
	activeMenu: Menu.DASHBOARD,
	activeSubMenu: '',
	plansMenuExpanded: false,
	backlogMenuExpanded: false,
};

export default function clickMenuReducer(state = initialState, action) {
	if (action.type === CLICK_MENU) {
		if (action.payload.menu === Menu.PLANNING && action.payload.subMenu === '') {
			return {
				activeMenu: action.payload.menu,
				activeSubMenu: action.payload.subMenu,
				plansMenuExpanded: !state.plansMenuExpanded,
				backlogMenuExpanded: state.backlogMenuExpanded ? false:state.backlogMenuExpanded
			};
		}
		else if (action.payload.menu === Menu.BACKLOG && action.payload.subMenu === '') {
			return {
				activeMenu: action.payload.menu,
				activeSubMenu: action.payload.subMenu,
				plansMenuExpanded: state.plansMenuExpanded ? false:state.plansMenuExpanded,
				backlogMenuExpanded: !state.backlogMenuExpanded
			};
		}
		return {
			activeMenu: action.payload.menu,
			activeSubMenu: action.payload.subMenu,
			plansMenuExpanded: state.plansMenuExpanded,
			backlogMenuExpanded: state.backlogMenuExpanded
		};
	}
	return state;
}
