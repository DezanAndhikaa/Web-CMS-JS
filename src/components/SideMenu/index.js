

import { connect } from 'react-redux';
import { push } from 'connected-react-router';
import { clickMenuAction, logoutAction } from './SideMenuComponent.actions';
import { toggleMenuAction } from '../NavigationBar/NavbarComponent.actions';
import { removeDataAction } from '../../core/StorageHelper';
import { USER_DATA } from '../../constants';
import SideMenuComponent from './SideMenuComponent';
import { serviceParameterAction, UpdateServiceParameterAction } from '../../views/planning/details/DetailPages-action';

const mapStateToProps = (state) => ({
	path: state.router.location.pathname,
	plansMenuExpanded: state.sideMenuComponentState.plansMenuExpanded,
	backlogMenuExpanded: state.sideMenuComponentState.backlogMenuExpanded,
	activeMenu: state.sideMenuComponentState.activeMenu,
	activeSubMenu: state.sideMenuComponentState.activeSubMenu,
	displayMode: state.displayMode,
	menuDrawerState: state.menuDrawerState,
	serviceParameter: state.plansPageState.serviceParameter,
	selectedFilters: state.plansPageState.selectedFilters,
	filterParameter: state.plansPageState.filterParameter
});

const mapDispatchToProps = (dispatch) => ({
	updateServiceParameter: (payload) =>
	  dispatch(serviceParameterAction(UpdateServiceParameterAction, payload)),
	clickMenu: (menu, subMenu) => dispatch(clickMenuAction({ menu, subMenu })),
	push: (path) => dispatch(push(path)),
	closeDrawer: () => dispatch(toggleMenuAction(false)),
	logout: () => dispatch(removeDataAction(USER_DATA)),
	onLogout: () => dispatch(logoutAction()),
});

export default connect(mapStateToProps, mapDispatchToProps)(SideMenuComponent);