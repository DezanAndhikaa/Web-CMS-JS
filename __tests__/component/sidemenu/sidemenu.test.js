import React from 'react';
import Enzyme, { shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import SideMenu from '../../../src/components/SideMenu/SideMenuComponent';

Enzyme.configure({adapter:new Adapter()});
it("render side menu correctly", () => {
    const wrapper = shallow (
        <SideMenu />
    )
    expect(wrapper).toMatchSnapshot();
})