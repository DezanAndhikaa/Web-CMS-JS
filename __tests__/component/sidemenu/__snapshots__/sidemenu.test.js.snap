// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render side menu correctly 1`] = `
<WithStyles(ForwardRef(Drawer))
  anchor="left"
  className="menu"
  classes={
    Object {
      "docked": "docked",
      "paper": "drawer-paper",
    }
  }
  onClose={[Function]}
  variant="temporary"
>
  <WithStyles(ForwardRef(List))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      className="menu-item"
      key="dashboard"
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemIcon))
        classes={
          Object {
            "root": "icon-root",
          }
        }
      >
        <img
          alt="assignment icon"
          className="item-icon"
          src={null}
        />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        classes={
          Object {
            "primary": "item-text",
            "root": "item-text",
          }
        }
        primary="Dashboard"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      className="menu-item"
      key="jobs"
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemIcon))
        classes={
          Object {
            "root": "icon-root",
          }
        }
      >
        <img
          alt="assignment icon"
          className="item-icon"
          src={null}
        />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        classes={
          Object {
            "primary": "item-text",
            "root": "item-text",
          }
        }
        primary="Tracking"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      className="menu-item"
      key="plans"
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemIcon))
        classes={
          Object {
            "root": "icon-root",
          }
        }
      >
        <img
          alt="assignment icon"
          className="item-icon"
          src={null}
        />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        classes={
          Object {
            "primary": "item-text",
            "root": "item-text",
          }
        }
        primary="Planning"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(Collapse))
      timeout="auto"
      unmountOnExit={true}
    >
      <WithStyles(ForwardRef(List))
        disablePadding={true}
      >
        <WithStyles(ForwardRef(ListItem))
          button={true}
          className="sub-menu"
          key="plans-assignment"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(ListItemIcon))
            classes={
              Object {
                "root": "icon-root",
              }
            }
          >
            <img
              alt="assignment icon"
              className="item-icon"
              src={null}
            />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            classes={
              Object {
                "primary": "item-text",
                "root": "item-text",
              }
            }
            primary="Dashboard"
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          className="sub-menu"
          key="jobs-report"
          onClick={[Function]}
        >
          <WithStyles(ForwardRef(ListItemIcon))
            classes={
              Object {
                "root": "icon-root",
              }
            }
          >
            <img
              alt="assignment icon"
              className="item-icon"
              src={null}
            />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            classes={
              Object {
                "primary": "item-text",
                "root": "item-text",
              }
            }
            primary="Site"
          />
        </WithStyles(ForwardRef(ListItem))>
      </WithStyles(ForwardRef(List))>
    </WithStyles(ForwardRef(Collapse))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      className="menu-item"
      key="jobs"
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemIcon))
        classes={
          Object {
            "root": "icon-root",
          }
        }
      >
        <img
          alt="assignment icon"
          className="item-icon"
          src={null}
        />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        classes={
          Object {
            "primary": "item-text",
            "root": "item-text",
          }
        }
        primary="Production"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      className="menu-item"
      key="jobs"
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemIcon))
        classes={
          Object {
            "root": "icon-root",
          }
        }
      >
        <img
          alt="assignment icon"
          className="item-icon"
          src={null}
        />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        classes={
          Object {
            "primary": "item-text",
            "root": "item-text",
          }
        }
        primary="Allocation"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      className="menu-item"
      key="jobs"
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemIcon))
        classes={
          Object {
            "root": "icon-root",
          }
        }
      >
        <img
          alt="assignment icon"
          className="item-icon"
          src={null}
        />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        classes={
          Object {
            "primary": "item-text",
            "root": "item-text",
          }
        }
        primary="Delivery"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      button={true}
      className="menu-item"
      key="jobs"
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemIcon))
        classes={
          Object {
            "root": "icon-root",
          }
        }
      >
        <img
          alt="assignment icon"
          className="item-icon"
          src={null}
        />
      </WithStyles(ForwardRef(ListItemIcon))>
      <WithStyles(ForwardRef(ListItemText))
        classes={
          Object {
            "primary": "item-text",
            "root": "item-text",
          }
        }
        primary="Execution"
      />
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(List))>
</WithStyles(ForwardRef(Drawer))>
`;
